isofit/radiative_transfer/radiative_transfer.py

```diff
@@ -167,80 +166,69 @@ def coszen(self):
                 return child.lut.coszen.data
 
     def calc_rdn(self, x_RT, x_surface, rfl, Ls, geom):
+        """
+        Physics-based forward model to calculate at-sensor radiance.
+        Includes topography, background reflectance, and glint.
+        """
+        # local solar zenith angle as a function of surface slope and aspect
+        cos_i = geom.cos_i if geom.cos_i is not None else self.coszen
+
+        # get needed rt quantities from LUT
         r = self.get_shared_rtm_quantities(x_RT, geom)
+
+        # atmospheric path radiance
         L_atm = self.get_L_atm(x_RT, geom)
-        L_up = Ls * (r["transm_up_dir"] + r["transm_up_dif"])
 
-        if geom.bg_rfl is not None:
-            # adjacency effects are counted
-            I = (self.solar_irr * self.coszen) / np.pi
-            bg = geom.bg_rfl
-            t_down = r["transm_down_dif"] + r["transm_down_dir"]
-
-            ret = (
-                L_atm
-                + I / (1.0 - r["sphalb"] * bg) * bg * t_down * r["transm_up_dif"]
-                + I / (1.0 - r["sphalb"] * bg) * rfl * t_down * r["transm_up_dir"]
-                + L_up
-            )
+        # atmospheric spherical albedo
+        s_alb = r["sphalb"]
 
-        elif self.topography_model:
-            I = self.solar_irr / np.pi
-            t_dir_down = r["transm_down_dir"]
-            t_dif_down = r["transm_down_dif"]
-            if geom.cos_i is None:
-                cos_i = self.coszen
-            else:
-                cos_i = geom.cos_i
-            t_total_up = r["transm_up_dif"] + r["transm_up_dir"]
-            t_total_down = t_dir_down + t_dif_down
-            s_alb = r["sphalb"]
-            # topographic flux (topoflux) effect corrected
-            ret = (
-                L_atm
-                + (
-                    I * cos_i / (1.0 - s_alb * rfl) * t_dir_down
-                    + I * self.coszen / (1.0 - s_alb * rfl) * t_dif_down
-                )
-                * rfl
-                * t_total_up
-            )
+        # direct and diffuse downward fluxes on the sun-to-surface path
+        # note: currently, E_down_dir comes scaled by the TOA solar zenith angle,
+        # thus, unscaling and rescaling by local solar zenith angle required
+        # to account for surface slope and aspect
+        E_down_dir, E_down_dif = self.get_E_down(x_RT, geom)
+        E_down_dir = E_down_dir / self.coszen * cos_i
```

Comment:
@pgbrodrick @davidraythompson When interpolation in a LUT based on the KF emulator or in Nimrod's universal LUT, I believe the obtained downward direct flux at the surface is already scaled by the TOA solar zenith angle, since that's the value we use for interpolating in the solar zenith angle LUT dimension. Therefore, I implemented this weird rescaling. Please let me know if I'm missing something here.

Reply:
Resolved as per conversation with the team in a previous tag up.